addons/website_sale/static/src/website_builder/dynamic_snippet_category_option_plugin.js

```diff
@@ -0,0 +1,120 @@
+import { rpc } from "@web/core/network/rpc";
+
+
+export class DynamicSnippetCategoryOptionPlugin extends Plugin {
+    static id = 'dynamicSnippetCategoryOptionPlugin';
+    selector = "section.s_dynamic_category"
+    resources = {
+        builder_options: [
+            withSequence(DEVICE_VISIBILITY, {
+                OptionComponent: DynamicSnippetCategoryOption,
+                selector: this.selector,
+                groups: ['website.group_website_designer'],
+            }),
+        ],
+        on_snippet_dropped_handlers: this.onSnippetDropped.bind(this),
+    };
+
+    async onSnippetDropped({ snippetEl }) {
+        if (snippetEl.matches(this.selector)) {
+            for (const [optionName, value] of [
+                ["columns", "2"],
+                ["height", "small"],
+                ["filterId", "0"],
+                ["allProducts", "true"],
+                ["button", "Explore Now"],
+                ["alignment", "left"],
+            ]) {
+                setDatasetIfUndefined(snippetEl, optionName, value);
+            }
+        }
+    }
+}
+
+export class CategoryMediaDialog extends MediaDialog {
+    async save(){
+        const nodeData = this.props.node.parentElement.dataset
+        rpc('/snippets/category/set_image',{
+            category_id: parseInt(nodeData.categoryId),
+            media: this.selectedMedia[this.state.activeTab],
+        })
+        this.props.close()
+        await super.save();
+    }
+}
+
+export class DynamicSnippetCategoryItemOptionPlugin extends Plugin {
+    static id = 'dynamicSnippetCategoryItemOptionPlugin';
+    static dependencies = ["media", "dialog", "builder-options"];
+    resources = {
+        builder_options: {
+            template: "website_sale.dynamicSnippetCategoryItemOptions",
+            selector: ".category_item, .all_products",
+            editableOnly: false,
+            title: _t("Category"),
+        },
+        builder_actions: this.getActions(),
+        patch_builder_options: [
+            {
+                target_name: 'websiteSaleRibbonOption',
+                target_element: 'selector',
+                method: 'add',
+                value: '.category_item',
+            },
+        ],
+    }
+    getActions() {
+        return {
+            setCategoryImage: {
+                reload: {},
+                load: async ({ editingElement: el }) => {
+                    const imageEl = el.querySelector(".s_category_image")
+                    if(el.classList.contains("category_item")){
+                        await new Promise((resolve) => {
+                            const onClose = this.dependencies.dialog.addDialog(CategoryMediaDialog, {
+                                node: imageEl,
+                                onlyImages: true,
+                                noDocuments: true,
+                                save: (icon) => resolve(icon),
+                            });
+                            onClose.then(resolve);
```

Comment:
```suggestion
                            }).then(resolve);
```
No need to declare `onClose` variable